scipy/ndimage/tests/test_measurements.py

```diff
@@ -1390,3 +1390,14 @@ def test_watershed_ift08(self):
         expected = [[1, 1],
                     [1, 1]]
         assert_array_almost_equal(out, expected)
+
+    def test_watershed_ift09(self):
+        # Test large cost. See gh-19575
+        data = np.array([[(1 << 16) - 1, 0],
+                         [0, 0]], np.uint16)
+        markers = np.array([[1, 0],
+                            [0, 0]], np.int8)
+        out = ndimage.watershed_ift(data, markers)
+        expected = [[1, 1],
+                    [1, 1]]
+        assert_array_almost_equal(out, expected)
```

Comment:
Maybe for *new* testing code we can use `assert_allclose` as the docstring of the testing function suggests? To be honest, `assert_array_equal` exact check passed for me locally on this branch, though I don't know offhand if the integer type dependence is safe/guaranteed for this case.

Reply:
I changed to `assert_allclose`. (Usually I use `assert_array_equal` for comparing integer arrays and it should be safe for to use for this test.)